cifar10_cnn/cifar_train.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * Simple Convolutional Neural Network for 
+ * object classification using CIFAR-10 dataset.
+ *
+ * @author David Port Louis
+ */
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/split_data.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <ensmallen.hpp>
+#include "periodic_save.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace arma;
+using namespace std;
+using namespace ens;
+
+// Utility function to generate class labels
+// from probabilities.
+arma::Row<size_t> getLabels(arma::mat yPreds) 
```

Comment:
```suggestion
arma::Row<size_t> getLabels(const arma::ma&t yPreds) 
```

To avoid an extra copy.